frontend/src/components/GenerateSchedulesUnitAndTimePicker.js

```diff
@@ -0,0 +1,184 @@
+import React,{useState} from 'react';
+import {Card, Form, Row, Button, Col, FormLabel, FormGroup, Modal} from 'react-bootstrap';
+import {Select} from "@material-ui/core";
+import InputLabel from '@material-ui/core/InputLabel';
+import { makeStyles } from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import FormControl from '@material-ui/core/FormControl';
+import Container from '@material-ui/core/Container';
+
+import {Label} from "@material-ui/icons";
+const useStyles = makeStyles((theme) => ({
+    formControl: {
+        margin: theme.spacing(1),
+        minWidth: 120,
+    },
+    selectEmpty: {
+        marginTop: theme.spacing(2),
+    },
+    container: {
+        display: 'flex',
+        flexWrap: 'wrap',
+    },
+    textField: {
+        marginLeft: theme.spacing(1),
+        marginRight: theme.spacing(1),
+        width: 200,
+    },
+}));
+
+
+export default function GenerateSchedulesUnitAndTimePicker() {
+    const classes = useStyles();
+    const [date, setDate,checkUnit] = useState('');
+    const[unit]=useState('')
+    let maxUnit,minUnit;
+    function setMin(e){
+        if(e===''){
+            maxUnit=12;
+        }else{
+            maxUnit=e;
+        }
+    }
+    function setMax(e){
+        if(e===''){
+            minUnit=19;
+        }else{
+            minUnit=e;
+        }
+    }
+
+
+    const handleChange = (event) => {
+        setDate(event.target.value);
+    };
+    async function handleSubmit(){
+        if(maxUnit>19||minUnit>19||maxUnit<12||minUnit<12||maxUnit<minUnit)
+        {
+            handleShow();
+        }
+    }
+
+    const [show, setShow] = React.useState(false);
+
+    const handleClose = () => setShow(false);
+    const handleShow = () => setShow(true);
+
+    return(
+        <>
+            <Modal show={show} onHide={handleClose} animation={false}>
+                <Modal.Header closeButton>
+                    <Modal.Title>Warning</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>Please type in number between 12 and 19 again</Modal.Body>
+                <Modal.Footer>
+                    <Button variant="secondary" onClick={handleClose}>
+                        Close
+                    </Button>
+                </Modal.Footer>
+            </Modal>
+<Container fixed>
+
+<Row>
+    <Col>
+                <Row>
```

Comment:
The whole filters component should be a form, so you can use Form.Row and Form.Col elements. Having each filter element be its own form isn't the best solution.